subsys/bluetooth/host/mesh/net.c

```diff
@@ -1179,14 +1179,26 @@ static void bt_mesh_net_relay(struct net_buf_simple *sbuf,
 		goto done;
 	}
 
-	if (IS_ENABLED(CONFIG_BT_MESH_GATT_PROXY)) {
+	/* Sending to the GATT bearer should only happen if GATT Proxy
+	 * is enabled or the message originates from the local node.
+	 */
+	if (IS_ENABLED(CONFIG_BT_MESH_GATT_PROXY) &&
+	    (bt_mesh_gatt_proxy_get() == BT_MESH_GATT_PROXY_ENABLED ||
+	     rx->net_if == BT_MESH_NET_IF_LOCAL)) {
 		if (bt_mesh_proxy_relay(&buf->b, rx->dst) &&
-			    BT_MESH_ADDR_IS_UNICAST(rx->dst)) {
+		    BT_MESH_ADDR_IS_UNICAST(rx->dst)) {
 			goto done;
 		}
 	}
 
-	bt_mesh_adv_send(buf, NULL, NULL);
+	/* Relaying to the Advertising bearer should only happen if the
+	 * Relay state is set to enabled, or if the packet has been
+	 * received over some other bearer than Advertising.
+	 */
+	if (bt_mesh_relay_get() == BT_MESH_RELAY_ENABLED ||
```

Comment:
I have feeling it was like that before and it was changed due to Friend support. Just wondering if it will not break anything else from Friend . However it looks good to me.

Reply:
If you mean us being the Friend and packets destined to the LPN, then those do not go through this function. They go through the Friend Queue and are given to bt_mesh_adv_send() directly from friend.c.

Reply:
Ah! IIRC an early incarnation of Friend support did indeed have some code in bt_mesh_net_relay(), however that was moved to transport.c since Friendship is a transport-layer feature. I.e. bt_mesh_net_recv() calls bt_mesh_trans_recv() which handles this, and only after it has returned does bt_mesh_net_relay() get called (which has nothing to do with Friendship anymore at this point).

Reply:
I'd also like to add that when we're a Friend, the packets that our LPN sends out have no special meaning to us, except if they're Friend control packets and/or we're subscribing to the destination address in such a packet. For any other packets from the LPN we have no extra obligations for relaying them compared to other mesh network packets.

Reply:
Ok, thanks for explanation.